core/common/src/main/java/alluxio/util/executor/ControllableScheduler.java

```diff
@@ -0,0 +1,311 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util.executor;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * A controllable {@link ScheduledExecutorService} that supports jump to a future time, runs
+ * next/all pending commands. This class is designed for test purpose.
+ */
+public class ControllableScheduler implements ScheduledExecutorService {
```

Comment:
Can we add a test for this?

We can reduce the work needed by throwing UnsupportedOperationException for the methods we don't currently use.

Reply:
Added the ControllableSchedulerTest, thanks